### PR TITLE
support "distinct(bla)" and "FROM foo, bar"

### PR DESCRIPTION
**Pull request by rizar**
With these 2 simple fixes we can already parse almost all Yelp dataset (except for the "neighborhood" query.

### CODE REVIEW

File: process_sql.py [hunk 1]
```diff
@@ -162,14 +176,14 @@ def tokenize(string):
         if toks[i] in vals:
             toks[i] = vals[toks[i]]
 
-    # find if there exists !=, >=, <=
-    eq_idxs = [idx for idx, tok in enumerate(toks) if tok == "="]
+    # find if there exists !=, >=, <=, <>`
```
Comment (RaymondLi0) [inline process_sql.py:179]: Just to make sure I understand: Is this block meant to ensure that these symbols are represented as a single token after the tokenization?

File: process_sql.py [hunk 2]
```diff
@@ -212,9 +235,20 @@ def parse_col(toks, start_idx, tables_with_alias, schema, default_tables=None):
...
         if tok in schema.schema[table]:
             key = table + "." + tok
-            return start_idx + 1, schema.idMap[key]
+            col_id = schema.idMap[key]
+
+    if col_id is None:
+        if tok == 'as':
+            raise DerivedFieldAliasError(toks[idx + 1])
+        else:
+            assert "Error col: {}".format(tok)
```
Comment (RaymondLi0) [inline process_sql.py]: shouldn't this raise an error?

File: process_sql.py [hunk 3]
```diff
@@ -440,12 +478,17 @@ def parse_from(toks, start_idx, tables_with_alias, schema):
...
+
         if toks[idx] == "select":
             idx, sql = parse_sql(toks, idx, tables_with_alias, schema)
             table_units.append((TABLE_TYPE["sql"], sql))
         else:
-            if idx < len_ and toks[idx] == "join":
+            if idx < len_ and toks[idx] in [",", "join"]:
                 idx += 1  # skip join
+            if idx + 1 < len_ and toks[idx:idx + 2] == ["inner", "join"]:
+                idx += 2  # skip join
```
Comment (RaymondLi0) [inline process_sql.py:491]: Should it be an `elif` here?
Otherwise this would allow incorrect SQL like `JOIN INNER JOIN` to be parsed correctly I think.

**Review by RaymondLi0 — COMMENTED** [review]
I like how we now have more precise types of errors. It's also nice to have tests!
The `.pyc` file should not be in there, maybe you can add a `.gitignore`. I also have a few comments here and there.

File: process_sql.py (hunk 1 above)
Comment (rizar) [reply process_sql.py:179]: Exactly!

File: process_sql.py (hunk 2 above)
Comment (rizar) [reply process_sql.py]: Yes, but they use assertions everywhere in the code, so I thought I'd stick to their style unless I am introduce a new, more specific error type.

File: process_sql.py (hunk 3 above)
Comment (rizar) [reply process_sql.py:491]: you are right, will fix that

File: process_sql.py (hunk 2 above)
Comment (RaymondLi0) [reply process_sql.py]: Oh ok I see, it should be `assert False, ...` then ?

**Comment by rizar** [comment]
Good catch!

**Review by RaymondLi0 — APPROVED** [review]
LGTM thanks!

**Pull request merged**